client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractHlrcWriteableXContentTestCase.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client;
+
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.test.AbstractSerializingTestCase;
+
+import java.io.IOException;
+
+import static org.elasticsearch.test.AbstractXContentTestCase.xContentTester;
+
+/**
+ * @deprecated Use {@link AbstractResponseTestCase} instead of this class.
+ */
+// TODO: Remove and change subclasses to use AbstractResponseTestCase instead
+@Deprecated
+public abstract class AbstractHlrcWriteableXContentTestCase<T extends ToXContent & Writeable, H>
```

Comment:
I introduced this class, because some hlrc tests that were changed in this PR extended from `AbstractHlrcStreamableXContentTestCase` and there was no `Writable` variant of this base test class.
I chose to introduce this class instead of changing the tests changed in the pr to extend from `AbstractResponseTestCase`, because changing hlrc tests to `AbstractResponseTestCase` is a refactoring of its own.